### PR TITLE
Add some "What are all these labels for?" documentation

### PR DESCRIPTION
**Pull request by rallytime**
Refs #23101

### CODE REVIEW

File: doc/topics/development/labels.rst [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+^^^^^^^^
+
+How severe is the bug? SaltStack uses four labels to determine the severity of a bug: ``Blocker``, ``Critical``,
+``High``, and ``Medium``. This scale is intended to make the bug-triage process as objective as possible.
+
+Blocker
+    Should be used sparingly to indicate must-have fixes for the impending release.
+
+Critical
+    Applied to bugs that have data loss, crashes, hanging, unresponsive system, or have no workaround.
```
Comment (basepi) [inline doc/topics/development/labels.rst]: I wouldn't mention workarounds here. We don't want to make any issue without a workaround Critical. *shrugs*

File: doc/topics/development/labels.rst [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+High Severity
+    Any bug report that contains incorrect functionality, bad functionality, a confusing user experience, or has a
+    possible workaround.
```
Comment (basepi) [inline doc/topics/development/labels.rst]: Same as above, probably just don't bother mentioning workarounds.

**Comment by basepi** [comment]
Looks great to me! You're awesome, @rallytime!

**Comment by rallytime** [comment]
Good call. Done. Thanks @basepi!

File: doc/topics/development/labels.rst [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+to the Salt engineering staff who should be reviewing and either discussing the changes with the contributor or merging
+the changes into the codebase.
+
+The change labels are listed in order of smallest/least critical code area change to largest/most critical area:
+
+* Minor Change
+* Medium Change
+* Master Change
+* Expert Change
+
```
Comment (jfindlay) [inline doc/topics/development/labels.rst]: I think we should describe these labels.

The conditions given for these labels are recommendations, as the reviewer will also consult their intuition and experience regarding the magnitude of the impact of the proposed changes in the pull request.

Core code areas include: state compiler, crypto engine, master and minion, transport, pillar rendering, loader, transport layer, event system, salt.utils, client, cli, logging, netapi, runner engine, templating engine, top file compilation, file client, file server, mine, salt-ssh, test runner

* Minor Change
  - < 64 lines changed
  - < 8 core lines changed
* Medium Change
  - < 256 lines changed
  - < 64 core lines changed
* Master Change
  - > 256 lines changed
  - > 64 core lines changed
* Expert Change
  - needs specialized, in-depth review

**Comment by rallytime** [comment]
Thanks @jfindlay! I've updated this to reflect those changes.

File: doc/topics/development/labels.rst [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+* Minor Change
+    * Less than 64 lines changed, or
+    * Less than 8 core lines changed
+* Medium Change
+    * Less than 256 lines changed, or
+    * Less than 64 core lines changed
+* Master Change
+    * More than 64 lines changed, or
+    * More than 256 core lines changed
```
Comment (jfindlay) [inline doc/topics/development/labels.rst]: Sorry to be picky but these should be reversed
    * More than 256 lines changed, or
    * More than 64 core lines changed

Comment (rallytime) [inline doc/topics/development/labels.rst]: Oh, yeah. Typo. Fixing now!

**Comment by lorengordon** [comment]
Very helpful, thanks! SaltStack has a great team!

**Pull request merged**